src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -1287,6 +1287,19 @@ void IVisualStudioWorkspaceHost.OnAdditionalDocumentTextUpdatedOnDisk(DocumentId
                 _workspace.OnAdditionalDocumentTextUpdatedOnDisk(id);
             }
 
+            void IVisualStudioWorkspaceHost.OnIntellisenseBuildResult(ProjectId projectId, bool succeed)
+            {
+                // this can't be null
+                var state = _workspace.CurrentSolution.GetProjectState(projectId);
+                Contract.ThrowIfNull(state);
+
+                // if state is different than what we have
+                if (succeed != state.ProjectInfo.IsComplete)
+                {
+                    _workspace.OnProjectReloaded(state.ProjectInfo.WithIsComplete(succeed));
```

Comment:
somewhat done. I still use OnProjectReloaded. if we add new OnXXX then we need to add new workspace changed kind and then need to audit ones that listen to workspace events and etc.